### ISSUE TITLE
Query Library frontend

### ISSUE THREAD

**Issue by mason-fish** [brimsec/brim#1081]
@orochford had some awesome suggestions for a query library in his "Initial Brim Query Use Cases" document. Let's implement them!

Some of the constraints we had discussed for this:
* The query lib would be a third section within the left side panel
* It would organize queries by type/category, each of which are collapsible
* Query libs can be imported/exported as .json through user prefs, and we will provide an initial lib as a sort of "getting started" template with common queries in it
* Queries will be annotated with a display name, which is what is shown in the library, and that name along with it's underlying query can also be edited/deleted
* Queries are space-agnostic: by selecting a given query, it will be applied to whatever current space you are on, or not at all if no space is open.

**Comment by jameskerr** [comment]
I'll add that these queries should also be connection and time span independent. But what about pins? I'd think if a user saved a query with pins, we should retain the pins?

**Comment by philrz** [comment]
Verified in Brim commit `77755a1`.

After importing test pcap https://archive.wrccdc.org/pcaps/2018/wrccdc.2018-03-23.010014000000000.pcap.gz, in the first attached video, I successfully click through all the built-in queries in the library.

https://user-images.githubusercontent.com/5934157/103104243-f0331a00-45da-11eb-8265-1fc6366213f9.mov

Then I add a new query via the "star" icon to the right of the search bar, tag it, and filter for it and another query via tag.

https://user-images.githubusercontent.com/5934157/103104264-0e007f00-45db-11eb-84c2-e219fe63692e.mov

Finally, I edit and ultimately delete the query I just added.

https://user-images.githubusercontent.com/5934157/103104279-1c4e9b00-45db-11eb-8429-f815345253cc.mov

Thanks @mason-fish!
